### PR TITLE
Colorize hyper rectangle with simplices

### PR DESCRIPTION
**Pull request by nmuch**

### CODE REVIEW

**Review by peterrum — APPROVED** [review]
/rebuild

**Review by kronbichler — APPROVED** [review]

File: source/grid/grid_generator.cc [hunk 1]
```diff
@@ -8782,11 +8780,28 @@ namespace GridGenerator
...
+    if (colorize)
+      {
+        // to colorize, run through all faces of all cells and set boundary
+        // indicator to the correct value if it was 0.
+
+        // use a large epsilon to compare numbers to avoid roundoff problems.
+        double epsilon = 10;
+        for (unsigned int i = 0; i < dim; ++i)
+          epsilon = std::min(epsilon,
+                             0.01 * (std::abs(p2[i] - p1[i]) / repetitions[i]));
```
Comment (bangerth) [inline source/grid/grid_generator.cc:8798]: This is not the right way to approach this. `p2[i] - p1[i]` is a quantity that depends on the magnitude of the points chosen as inputs, so it has physical units. But `eps=10` does not. You need to start with an eps that has the right order of magnitude. Alternatively, you could start with `eps=infinity`, i.e., `eps=std::numeric_limits<double>::max()`.

**Review by bangerth — CHANGES_REQUESTED** [review]

**Review by kronbichler — APPROVED** [review]
Thanks for the fix, and thanks to @bangerth for noticing!

**Review by masterleinad — APPROVED** [review]

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
This looks right now. Thank you!

It would be nice to have a test that checks this. Do you know how to clone existing tests and add those?

**Review by kronbichler — APPROVED** [review]

**Pull request merged**